src/cljam/algo/convert.clj

```diff
@@ -11,20 +13,24 @@
 (def ^:private default-num-block 100000)
 
 (defn- sam-write-alignments [rdr wtr hdr num-block]
+  (when (and (pos? *n-threads*) (> (get-exec-n-threads) 1))
+    (logging/warn "Concurrent SAM writing is not supported."))
   (doseq [alns (partition-all num-block (sam/read-alignments rdr {}))]
     (sam/write-alignments wtr alns hdr)))
 
 (defn- bam-write-alignments [rdr wtr hdr num-block]
-  (let [refs (sam-util/make-refs hdr)]
-    (cp/with-shutdown! [pool (cp/threadpool (cp/ncpus))]
-      (doseq [blocks (cp/pmap pool
-                              (fn [chunk]
-                                (mapv #(let [bb (ByteBuffer/allocate (encoder/get-block-size %))]
-                                         (encoder/encode-alignment bb % refs)
-                                         {:data (.array bb)})
-                                      chunk))
-                              (partition-all num-block (sam/read-alignments rdr {})))]
-        (sam/write-blocks wtr blocks)))))
+  (let [refs (sam-util/make-refs hdr)
+        n-threads (get-exec-n-threads)
+        map-fn (if (= n-threads 1)
+                 map
+                 (partial cp/pmap (cp/threadpool (dec n-threads))))]
```

Comment:
I noticed that claypoole takes care of [switching map/pmap](https://github.com/TheClimateCorporation/claypoole/blob/9a37bfd312742ddd48b7c146c5e5bb9e924d1a52/src/clj/com/climate/claypoole.clj#L368) and [creating a temporal thread pool](https://github.com/TheClimateCorporation/claypoole/blob/9a37bfd312742ddd48b7c146c5e5bb9e924d1a52/src/clj/com/climate/claypoole/impl.clj#L227).

So it's equivalent to

```clojure
(partial cp/pmap (if (= n-threads 1) :serial (dec n-threads)))
```